packages/core/components/DirectoryTree/findChildNodes.ts

```diff
@@ -0,0 +1,113 @@
+import { defaults, isEmpty, pull } from "lodash";
+
+import { NO_VALUE_NODE, ROOT_NODE } from "./directory-hierarchy-state";
+import FileFilter, { FilterType } from "../../entity/FileFilter";
+import FileSet from "../../entity/FileSet";
+import { AnnotationService } from "../../services";
+import { naturalComparator } from "../../util/strings";
+
+export interface FindChildNodesParams {
+    ancestorNodes?: string[];
+    currentNode: string;
+    hierarchy: string[];
+    annotationService: AnnotationService;
+    fileSet?: FileSet;
+    selectedFileFilters?: FileFilter[];
+    shouldShowNullGroups?: boolean;
+}
+
+const DEFAULTS = {
+    ancestorNodes: [],
+    currentNode: ROOT_NODE,
+    selectedFileFilters: [],
+    shouldShowNullGroups: false,
+};
+
+export async function findChildNodes(params: FindChildNodesParams): Promise<string[]> {
+    const {
+        ancestorNodes,
+        currentNode,
+        fileSet,
+        hierarchy,
+        annotationService,
+        selectedFileFilters,
+        shouldShowNullGroups,
+    } = defaults({}, params, DEFAULTS);
+
+    const isRoot = currentNode === ROOT_NODE;
+
+    const combinedFileFilters = selectedFileFilters.concat(
+        (fileSet?.filters || [])?.filter((filter) => !selectedFileFilters.includes(filter))
+    );
+
+    // If at root of hierarchy, currentNode will be set to "ROOT_NODE"
+    // We trim that from the path as it is not meaningful in this context
+    const pathToNode = pull([...ancestorNodes, currentNode], ROOT_NODE);
+
+    let values: string[] = [];
+
+    const depth = pathToNode.length;
+    const annotationNameAtDepth = hierarchy[depth];
+    const userSelectedFiltersForCurrentAnnotation = combinedFileFilters
+        .filter(
+            (filter) => filter.name === annotationNameAtDepth && filter.type !== FilterType.EXCLUDE
+        )
+        .map((filter) => filter.value);
+
+    if (isRoot) {
```

Comment:
I think this could be written like so to simplify the ternary operator check:
```
if (shouldShowNullGroups) {
        // Fetch all values under node, ignoring past hierarchy
        values = await annotationService.fetchRootHierarchyValues(
            [annotationNameAtDepth],
            combinedFileFilters
        );
} else if (isRoot) {
         values = await annotationService.fetchRootHierarchyValues(
            hierarchy,
            combinedFileFilters
        );
} else {
        values = await annotationService.fetchHierarchyValuesUnderPath(
            hierarchy,
            pathToNode,
            combinedFileFilters
        );
}
```